network/network_linux.go

```diff
@@ -292,10 +303,23 @@ func (nm *networkManager) applyIPConfig(extIf *externalInterface, targetIf *net.
 }
 
 func applyDnsConfig(extIf *externalInterface, ifName string) error {
-	var err error
+	var (
+		setDnsList string
+		err        error
+	)
+
+	if extIf != nil {
+		for _, server := range extIf.DNSInfo.Servers {
+			if net.ParseIP(server).To4() == nil {
```

Comment:
I think Azure does support ipv6 DNS servers, is that true for VM DNS servers ? [Azure-DNS-Ipv6](https://docs.microsoft.com/en-us/azure/dns/dns-faq#do-azure-dns-name-servers-resolve-over-ipv6)

Reply:
imo can't add ipv6 dns server in v4 vnet. we can consider this when we support dualstack